app/components/code-edit-modal.js

```diff
@@ -0,0 +1,9 @@
+import Component from '@glimmer/component';
+import { restartableTask, timeout } from 'ember-concurrency';
+
+export default class CodeEditModal extends Component {
+  handleCodeChange = restartableTask(async (newCode) => {
+    await timeout(500);
```

Comment:
Moved the long timeout here. We still want to prevent multiple events while the user is still typing.

Reply:
Yes indeed I don't see any other tasks that could fix this.. Maybe the `keepLatest` task but it's just going to make everything slower BUT insure that every change in the store is added piece by piece (ignore after the but if its not clear :)) https://ember-concurrency.com/docs/task-concurrency